sonar-application/src/main/java/org/sonar/application/JavaCommandFactoryImpl.java

```diff
@@ -0,0 +1,114 @@
+import java.io.File;
+import org.sonar.process.ProcessId;
+import org.sonar.process.ProcessProperties;
+import org.sonar.process.Props;
+import org.sonar.process.monitor.JavaCommand;
+
+import static org.sonar.process.ProcessProperties.HTTPS_PROXY_HOST;
+import static org.sonar.process.ProcessProperties.HTTPS_PROXY_PORT;
+import static org.sonar.process.ProcessProperties.HTTP_PROXY_HOST;
+import static org.sonar.process.ProcessProperties.HTTP_PROXY_PORT;
+
+public class JavaCommandFactoryImpl implements JavaCommandFactory {
+  /**
+   * Properties about proxy that must be set as system properties
+   */
+  private static final String[] PROXY_PROPERTY_KEYS = new String[] {
+    HTTP_PROXY_HOST,
+    HTTP_PROXY_PORT,
+    "http.nonProxyHosts",
+    HTTPS_PROXY_HOST,
+    HTTPS_PROXY_PORT,
+    "http.auth.ntlm.domain",
+    "socksProxyHost",
+    "socksProxyPort"};
+
+  @Override
+  public JavaCommand createESCommand(Props props, File workDir) {
+      return newJavaCommand(ProcessId.ELASTICSEARCH, props, workDir)
+        .addJavaOptions("-Djava.awt.headless=true")
+        .addJavaOptions(props.nonNullValue(ProcessProperties.SEARCH_JAVA_OPTS))
+        .addJavaOptions(props.nonNullValue(ProcessProperties.SEARCH_JAVA_ADDITIONAL_OPTS))
+        .setClassName("org.sonar.search.SearchServer")
+        .addClasspath("./lib/common/*")
+        .addClasspath("./lib/search/*");
+    }
+
+  @Override
+  public JavaCommand createWebCommand(Props props, File workDir) {
+      JavaCommand command = newJavaCommand(ProcessId.WEB_SERVER, props, workDir)
+        .addJavaOptions(ProcessProperties.WEB_ENFORCED_JVM_ARGS)
+        .addJavaOptions(props.nonNullValue(ProcessProperties.WEB_JAVA_OPTS))
+        .addJavaOptions(props.nonNullValue(ProcessProperties.WEB_JAVA_ADDITIONAL_OPTS))
+        // required for logback tomcat valve
+        .setEnvVariable(ProcessProperties.PATH_LOGS, props.nonNullValue(ProcessProperties.PATH_LOGS))
+        .setClassName("org.sonar.server.app.WebServer")
+        .addClasspath("./lib/common/*")
+        .addClasspath("./lib/server/*");
+      String driverPath = props.value(ProcessProperties.JDBC_DRIVER_PATH);
+      if (driverPath != null) {
+        command.addClasspath(driverPath);
+      }
+      return command;
+    }
+
+  @Override
+  public JavaCommand createCeCommand(Props props, File workDir) {
+      JavaCommand command = newJavaCommand(ProcessId.COMPUTE_ENGINE, props, workDir)
+        .addJavaOptions(ProcessProperties.CE_ENFORCED_JVM_ARGS)
+        .addJavaOptions(props.nonNullValue(ProcessProperties.CE_JAVA_OPTS))
+        .addJavaOptions(props.nonNullValue(ProcessProperties.CE_JAVA_ADDITIONAL_OPTS))
+        .setClassName("org.sonar.ce.app.CeServer")
+        .addClasspath("./lib/common/*")
+        .addClasspath("./lib/server/*")
+        .addClasspath("./lib/ce/*");
+      String driverPath = props.value(ProcessProperties.JDBC_DRIVER_PATH);
+      if (driverPath != null) {
+        command.addClasspath(driverPath);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Make this line start at column 7. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AIndentationCheck)